### PR TITLE
Add YOLOv4

### PR DESCRIPTION
**Pull request by gabrieltseng**
- [x] Add YOLO model
- [x] Clip the bounding boxes to ensure they are within the image
- [x] Allow models to define their own learning rate

### CODE REVIEW

File: benchmarks/src/models/architectures/ssd.py [hunk 1]
```diff
@@ -15,11 +15,14 @@ def initialize_ssd(
...
-) -> nn.Module:
-    return ssd300_vgg16(
-        pretrained=pretrained,
-        trainable_backbone_layers=trainable_backbone_layers,
-        num_classes=num_classes,
-        pretrained_backbone=pretrained_backbone,
-        **kwargs,
+) -> Tuple[nn.Module, float]:
+    return (
+        ssd300_vgg16(
```
Comment (TomW1495) [inline benchmarks/src/models/architectures/ssd.py:20]: Can this not be saved as and returned as "model" like the others?

File: benchmarks/src/models/architectures/yolo/__init__.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        max_predictions_per_image=max_predictions_per_image,
+        image_height=model_config.global_config["height"],
+        image_width=model_config.global_config["width"],
+    )
+    if pretrained:
+        if not DEFAULT_WEIGHT_PATH.exists():
+            download_from_url(YOLOv4_DARKNET_WEIGHTS, DEFAULT_WEIGHT_PATH)
+        with DEFAULT_WEIGHT_PATH.open() as weight_file:
+            load_darknet_weights(weight_file, model)
+    return model, 0.001
```
Comment (TomW1495) [inline benchmarks/src/models/architectures/yolo/__init__.py]: Instead of returning the default LR as a hard-coded value at the bottom. Could we have it as input to the initialize yolo for example? And set the default value there? I think it's a bit ambiguous left as is

Comment (TomW1495) [reply benchmarks/src/models/architectures/yolo/__init__.py]: Or somewhere else if that doesn't work

**Review by TomW1495 — APPROVED** [review]

File: benchmarks/src/models/architectures/ssd.py (hunk 1 above)
Comment (gabrieltseng) [reply benchmarks/src/models/architectures/ssd.py:20]: All the other models require something to happen to the `model` object; for SSD, there's no modifications necessary, so there is no need to create the extra variable.

Do you reckon it increases readability to do it with the `model` object? If so, I am happy to create it.

File: benchmarks/src/models/architectures/yolo/__init__.py (hunk 2 above)
Comment (gabrieltseng) [reply benchmarks/src/models/architectures/yolo/__init__.py]: The user has an option to override the default in the [model class](https://github.com/SmallRobotCompany/smallssd/blob/d2eea82818a0c2ccdcebb750c81a20443707fbcf/benchmarks/src/models/base.py#L16).

I've also created a `DEFAULT_YOLO_LR` variable to make it more explicit what's being returned

File: benchmarks/src/models/architectures/ssd.py (hunk 1 above)
Comment (TomW1495) [reply benchmarks/src/models/architectures/ssd.py:20]: If it's not needed then I think the readability is fine without

File: benchmarks/src/models/architectures/yolo/__init__.py (hunk 2 above)
Comment (TomW1495) [reply benchmarks/src/models/architectures/yolo/__init__.py]: Looks good to me!
